File: apps/ios/Pebbles/Services/AppEnvironment.swift
import Foundation

/// Typed access to build-time configuration values injected via
/// `Config/Secrets.xcconfig` → `Info.plist`. Fails loud and early if
/// a value is missing so setup bugs don't become runtime mysteries.
enum AppEnvironment {
    static let supabaseURL: URL = {
        guard let raw = Bundle.main.object(forInfoDictionaryKey: "SupabaseURL") as? String,
              !raw.isEmpty,
              let url = URL(string: raw) else {
            fatalError(
                "SupabaseURL missing or invalid in Info.plist. " +
                "Did you copy Config/Secrets.example.xcconfig → Config/Secrets.xcconfig?"
            )
        }
        return url
    }()

    static let supabaseAnonKey: String = {
        guard let key = Bundle.main.object(forInfoDictionaryKey: "SupabaseAnonKey") as? String,
              !key.isEmpty else {
            fatalError(
                "SupabaseAnonKey missing in Info.plist. " +
                "Did you copy Config/Secrets.example.xcconfig → Config/Secrets.xcconfig?"
            )
        }
        return key
    }()

    static let googleIOSClientID: String = {
        guard let id = Bundle.main.object(forInfoDictionaryKey: "GoogleIOSClientID") as? String,
              !id.isEmpty else {
            fatalError(
                "GoogleIOSClientID missing in Info.plist. " +
                "Did you set GOOGLE_IOS_CLIENT_ID in Config/Secrets.xcconfig?"
            )
        }
        return id
    }()
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/AppEnvironment.kt
package app.pbbls.android

/**
 * Typed access to build-time configuration injected via `secrets.properties` →
 * `BuildConfig` (see `app/build.gradle.kts`). Fails loud and early if a value is
 * missing so setup bugs don't become runtime mysteries — the Android analog of
 * the iOS `AppEnvironment` `fatalError` contract. A missing `secrets.properties`
 * never breaks the build; it surfaces here, at first access, instead.
 */
object AppEnvironment {
    val supabaseUrl: String
        get() = requireSecret("SUPABASE_URL", BuildConfig.SUPABASE_URL)

    val supabaseAnonKey: String
        get() = requireSecret("SUPABASE_ANON_KEY", BuildConfig.SUPABASE_ANON_KEY)

    /**
     * Returns [value] when it is non-blank, otherwise throws with copy-the-example
     * setup instructions. Extracted from the accessors (rather than reading
     * `BuildConfig` inline) so it can be unit-tested without a real
     * `secrets.properties` present on the machine.
     */
    internal fun requireSecret(
        name: String,
        value: String,
    ): String {
        check(value.isNotBlank()) {
            "$name is missing. Copy apps/android/secrets.example.properties to " +
                "secrets.properties and fill in real values."
        }
        return value
    }
}
